vm/actor/src/util/balance_table.rs

```diff
@@ -55,22 +55,22 @@ where
     /// Sets the balance for the address, overwriting previous value
     #[inline]
     pub fn set(&mut self, key: &Address, value: TokenAmount) -> Result<(), Error> {
-        self.0.set(key.hash_key(), BigUintDe(value))
+        self.0.set(key.to_bytes().into(), BigUintDe(value))
     }
 
     /// Adds token amount to previously initialized account.
     pub fn add(&mut self, key: &Address, value: &TokenAmount) -> Result<(), String> {
         let prev = self.get(key)?;
-        Ok(self.0.set(key.hash_key(), BigUintDe(prev + value))?)
+        Ok(self.0.set(key.to_bytes().into(), BigUintDe(prev + value))?)
     }
 
     /// Adds an amount to a balance. Creates entry if not exists
     pub fn add_create(&mut self, key: &Address, value: TokenAmount) -> Result<(), String> {
-        let new_val = match self.0.get::<_, BigUintDe>(&key.hash_key())? {
+        let new_val = match self.0.get::<_, BigUintDe>(&key.to_bytes())? {
             Some(v) => v.0 + value,
             None => value,
         };
-        Ok(self.0.set(key.hash_key(), BigUintDe(new_val))?)
+        Ok(self.0.set(key.to_bytes().into(), BigUintDe(new_val))?)
```

Comment:
you could just implement `impl<T: AsRef<str>> From<T> for BytesKey` and avoid all these `to_bytes` calls

Reply:
hmm, I don't think this helps, I'm not sure if you're seeing that the key is of the address type, it is being converted into a `Vec<u8>`  and then the `into()` is putting it as the wrapper type used as a key here (`BytesKey`) which is specifically needed to ensure the string serialization of those bytes

Reply:
Ah sorry, `.to_bytes()` made me think it was a string,